app/models/broadcast_message.py

```diff
@@ -271,6 +272,29 @@ def add_areas(self, *new_area_ids):
         self.area_ids = list(OrderedSet(self.area_ids + list(new_area_ids)))
         self._update_areas()
 
+    def add_custom_areas(self, *circle_polygon, id):
+        simple_polygons = list(circle_polygon)
+        local_authority = CustomBroadcastArea(name="", polygons=simple_polygons).local_authority
+        if local_authority:
+            id = f"{id}, in {local_authority}"
+        if id not in self.area_ids:
+            areas = {}
+            areas["ids"] = [id]
+            areas["names"] = [id]
+            areas["aggregate_names"] = [id]
+            if os.environ.get("IN_CICD"):
+                areas["simple_polygons"] = [
+                    [[round(coord, 10) for coord in polygon] for polygon in polygons] for polygons in simple_polygons
+                ]
+            else:
+                areas["simple_polygons"] = simple_polygons
```

Comment:
Is there a particular reason we're doing things differently for the tests in this file, it feels like we're masking a bug here.

Reply:
This is because the GitHub test runner is failing with a really minute difference in actual and expected coordinates. It seems like a floating point precision issue just within the GitHub runner instances and doesn't have any behavioural impact.

Reply:
This feels like something we should dig into and resolve, I'm not keen on having code in the libraries which bypasses code.  I think that you are right and it's a floating point issue and we can probably resolve it by either massaging the test data or by being more relaxed when we're inspecting the results inside the test test results.

Reply:
The test now loops through expected coordinates and actual and checks that they're within a tolerance of 1e-12 of each other. This is now passing within Github runner instances also, whereas it previously didn't without the rounding fix.